commit 80d13804d754f9ab9ab0f4bbcfd007fe383c3c05 (refs/changes/83/783/3)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2016-11-25T12:54:45+00:00 (2 years, 10 months ago)
    
    TOOLS-1581 `sdcadm post-setup common-external-nics` should update the SAPI service params.networks

diff --git a/lib/sdcadm.js b/lib/sdcadm.js
index f61f90d..4d72903 100644
--- a/lib/sdcadm.js
+++ b/lib/sdcadm.js
@@ -6132,6 +6132,34 @@ setupCommonExternalNics(opts, cb) {
         });
     }
 
+    function updateSvcParamsNetworks(name, subcb) {
+        self.sapi.listServices({name: name}, function (sapiErr, svcArr) {
+            if (sapiErr) {
+                subcb(new errors.SDCClientError(sapiErr, 'sapi'));
+                return;
+            }
+
+            if (!svcArr.length) {
+                subcb(new errors.InternalError(new Error(
+                    'Cannot find service ' + name)));
+                return;
+            }
+
+
+            self.sapi.updateService(svcArr[0].uuid, {
+                params: {
+                    networks: ['admin', 'external']
+                }
+            }, function (err) {
+                if (err) {
+                    subcb(new errors.SDCClientError(err, 'sapi'));
+                    return;
+                }
+                subcb();
+            });
+        });
+    }
+
     vasync.pipeline({ funcs: [
         function (_, next) {
             self.checkMissingExternalNics(opts, function (err, res) {
@@ -6198,7 +6226,7 @@ setupCommonExternalNics(opts, cb) {
                     return next(err);
                 }
                 progress('Added external nic to adminui');
-                next();
+                updateSvcParamsNetworks('adminui', next);
             });
         },
 
@@ -6212,7 +6240,7 @@ setupCommonExternalNics(opts, cb) {
                     return next(err);
                 }
                 progress('Added external nic to imgapi');
-                next();
+                updateSvcParamsNetworks('imgapi', next);
             });
         }
     ]}, function (err) {
diff --git a/test/post-setup.test.js b/test/post-setup.test.js
index d6cfc51..beee5be 100644
--- a/test/post-setup.test.js
+++ b/test/post-setup.test.js
@@ -429,6 +429,23 @@ test('teardown', function (t) {
         },
         inputs: vmsWithExternalNics
     }, function (resErr) {
-        t.end();
+        vasync.forEachParallel({
+            func: function updateParamsNetworks(svc, next) {
+                var command = 'echo \'{"params": {"networks": ["admin"]}}\'|' +
+                    'sapiadm update $(sdc-sapi /services?name=' + svc +
+                    '|json -Ha uuid)';
+                exec(command, function (err, stdout, stderr) {
+                    t.ifError(err, 'Execution error');
+                    t.equal(stderr, '', 'Empty stderr');
+                    next();
+                });
+
+            },
+            inputs: vmsWithExternalNics.map(function (vm) {
+                return vm.tags.smartdc_role;
+            })
+        }, function (paraRes) {
+            t.end();
+        });
     });
 });
